source/binary.cpp

```diff
@@ -732,6 +739,43 @@ spv_result_t Parser::setNumericTypeInfoForType(
   return SPV_SUCCESS;
 }
 
+spv_result_t Parser::verifyLiteralUpperBits(
+    const spv_parsed_operand_t& parsed_operand, uint32_t word) {
+  // Verify that the upper bits are either set to 0, or sign-extended
+  auto verify = [&](uint32_t mask, std::vector<uint32_t> valid_patterns)
+      -> spv_result_t {
+    uint32_t upper = word & mask;
+    for (uint32_t valid : valid_patterns) {
+      if (upper == valid) {
+        return SPV_SUCCESS;
+      }
+    }
+    return diagnostic() << "Upper bits of literal number are invalid";
+  };
+
+  switch (parsed_operand.number_kind) {
+    case SPV_NUMBER_SIGNED_INT:
+      switch (parsed_operand.number_bit_width) {
+        case 8:  return verify(0xFFFFFF00,{0x00000000,0xFFFFFF00});
```

Comment:
Generic SPIR-V can have any-width numeric types.  So this code is too specific.
It should work with 1-bit , 12-bit, 51-bit integers, for example.

So far we've limited SPIRV-Tools to handle any integer bit width up to and including 64 bits.

For floats, we only handle 16, 32, and 64 bit floats.